package com.bringmyweb.exceptionalframwork.core;

import com.bringmyweb.exceptionalframwork.annotations.CatchGroup;
import com.bringmyweb.exceptionalframwork.annotations.Catch;

class Foo implements Caller {
	@Override
	@CatchGroup(catchers = { @Catch(targetCatchHandler = Bar.class, targetException = ArithmeticException.class),
			@Catch(targetCatchHandler = Bar.class, targetException = NullPointerException.class) })
	public void callMethod() throws Throwable {
		int a = 0, b = 10;
		int c = (b / a);
	}

	public static void main(String[] args) throws Exception {
		Foo foo = new Foo();
		MethodCaller.callMethod(foo);
	}
}